src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1557,14 +1598,24 @@ static int client_parse_message(
                 assert_not_reached();
         }
 
-        lease->next_server = message->siaddr;
+        if (client->bootp) {
+                lease->lifetime = USEC_INFINITY;
+
+                log_dhcp_client(client, "BOOTP identified, using infinite lease. BOOTP siaddr=(%#x), DHCP Server Identifier=(%#x)",
+                        message->siaddr,
+                        lease->server_address);
+
+                lease->server_address = message->siaddr || lease->server_address;
```

Comment:
in systemd we do not use numeric variables like booleans unless they are actually boolean-like. moreover, what is server_address set to 1 even mean?

Reply:
Whoops, that wasn't supposed to be here. Should read like this
```suggestion
                lease->server_address = lease->server_address ? lease->server_address :  message->siaddr;
```

Reply:
Sorry about that @colin-foster-in-advantage , that was from my first iteration, when I forgot that C logical operators don't work like their scripting language counterparts.